### ISSUE TITLE
Dynamically updating semester event data

### ISSUE THREAD

**Issue by nur-zaman** [nur-zaman/better-usis#4]
currently the semester events data is hard coded in [data.ts](https://github.com/nur-zaman/better-usis/blob/master/constants/data.ts)
need to make it dynamic by using [https://www.bracu.ac.bd/academic/spring/2023/rss.xml](https://www.bracu.ac.bd/academic/spring/2023/rss.xml) .

**Comment by NaiveInvestigator** [comment]
uH sorry for some reason i can't run the project locally on my machine.

Steps I took:
```
git clone https://github.com/nur-zaman/better-usis
cd better-usis
npm install
npm run dev
```

The site starts up on localhost as expected but then when i try put my credentials it shows me this error:
![image](https://github.com/nur-zaman/better-usis/assets/72654596/773cffc3-0a46-4434-ab7f-ba073495914b)


I'm not so sure what to do next. What i really wanted to do is just run the app and see if my code works lol. Maybe i am doing something wrong. I've never worked with next.js so pls bear with me

Midterms ended now so was planning to take a crack at this issue.

Edit: Forgot to add logs mb, i redacted some info cause of obvious reasons

```
❯ npm run dev

> better-usis@0.1.0 dev
> next dev

   ▲ Next.js 14.1.0
   - Local:        http://localhost:3000

Attention: Next.js now collects completely anonymous telemetry regarding usage.
This information is used to shape Next.js' roadmap and prioritize features.
You can learn more, including how to opt-out if you'd not like to participate in this anonymous program, by visiting the following URL:
https://nextjs.org/telemetry

 ✓ Ready in 2.3s
(node:14686) [DEP0040] DeprecationWarning: The `punycode` module is deprecated. Please use a userland alternative instead.
(Use `node --trace-deprecation ...` to show where the warning was created)
 ○ Compiling / ...
 ✓ Compiled / in 5.8s (917 modules)
[next-auth][warn][NEXTAUTH_URL] 
https://next-auth.js.org/warnings#nextauth_url
[next-auth][warn][NO_SECRET] 
https://next-auth.js.org/warnings#no_secret
 ✓ Compiled in 438ms (324 modules)
 ○ Compiling /favicon.ico ...
 ✓ Compiled /favicon.ico in 2.8s (922 modules)
[next-auth][warn][NEXTAUTH_URL] 
https://next-auth.js.org/warnings#nextauth_url
[next-auth][warn][NO_SECRET] 
https://next-auth.js.org/warnings#no_secret
 ○ Compiling /api/auth/[...nextauth] ...
 ✓ Compiled /api/auth/[...nextauth] in 530ms (733 modules)
[next-auth][warn][NEXTAUTH_URL] 
https://next-auth.js.org/warnings#nextauth_url
[next-auth][warn][NO_SECRET] 
https://next-auth.js.org/warnings#no_secret
(node:15105) [DEP0040] DeprecationWarning: The `punycode` module is deprecated. Please use a userland alternative instead.
(Use `node --trace-deprecation ...` to show where the warning was created)
{
  email: '********@gmail.com',
  password: '********',
  callbackUrl: '/dashboard',
  csrfToken: '*********************************************************************',
  json: 'true'
}
(node:15123) [DEP0040] DeprecationWarning: The `punycode` module is deprecated. Please use a userland alternative instead.
(Use `node --trace-deprecation ...` to show where the warning was created)
 ○ Compiling /middleware ...
 ✓ Compiled /middleware in 528ms (170 modules)
[next-auth][error][NO_SECRET] 
https://next-auth.js.org/errors#no_secret
(node:15137) [DEP0040] DeprecationWarning: The `punycode` module is deprecated. Please use a userland alternative instead.
(Use `node --trace-deprecation ...` to show where the warning was created)
^C

```

**Comment by NaiveInvestigator** [comment]
*Sigh* nvm i forgot to do this: https://github.com/nur-zaman/better-usis/issues/3#issuecomment-1987055056
it works now :>

**Comment by nur-zaman** [comment]
Feel free to reach out if you face any other issues or confusions

**Comment by nur-zaman** [comment]
Merged with #8
